docs/tutorials/join.md

```diff
@@ -183,14 +178,49 @@ produces
 {name:"apple",color:"red",flavor:"tart",eater:"morgan"}
 ```
 
+## Alternate Syntax
+
+In addition to the syntax shown so far, `join` supports an alternate syntax in
+which left and right inputs are specified by the two legs of a preceeding
+[`fork` operator](../language/operators/fork.md),
+[`from` operator](../language/operators/from.md), or
+[`switch` operator](../language/operators/switch.md).
+
+Here we'll use the alternate syntax to perform the same inner join shown earlier
+in the [Inner Join section](#inner-join).
+
+The Zed script `inner-join-alternate.zed`:
+```mdtest-input inner-join-alternate.zed
+from (
+  file fruit.ndjson => sort flavor
+  file people.ndjson => sort likes
+) | inner join on flavor=likes eater:=name
+```
+
+Executing the Zed script:
+```mdtest-command
+zq -z -I inner-join-alternate.zed
+```
+produces
+```mdtest-output
+{name:"figs",color:"brown",flavor:"plain",eater:"jessie"}
+{name:"banana",color:"yellow",flavor:"sweet",eater:"quinn"}
+{name:"strawberry",color:"red",flavor:"sweet",eater:"quinn"}
+{name:"dates",color:"brown",flavor:"sweet",note:"in season",eater:"quinn"}
+{name:"apple",color:"red",flavor:"tart",eater:"morgan"}
+{name:"apple",color:"red",flavor:"tart",eater:"chris"}
+```
+
 ## Self Joins
 
 In addition to the named files and pools like we've used in the prior examples,
-Zed is also intended to work on a single sequence of data that is split
-and joined to itself.  Here we'll combine our file
-sources into a stream that we'll pipe into `zq` via stdin. Because `join` requires
-two separate inputs, here we'll use the `has()` function to identify the
-records in the stream that will be treated as the left and right sides.
+Zed is also intended to work on a single sequence of data that is split and
+joined to itself.  Here we'll combine our file sources into a stream that we'll
+pipe into `zq` via stdin.  Because `join` requires two separate inputs, here
+we'll use the `has()` function inside a `switch` operator to identify the
+records in the stream that will be treated as the left and right sides.  Then
+we'll use the [alternate syntax for `join`](#alternate-syntax) to read those two
+inputs.
```

Comment:
Am I remembering correctly that this example can only currently be expressed using this alternate syntax? If so, further along the lines of my suggestion in the `join` operator doc about calling attention to the two syntaxes, I'd be a fan of putting some explicit acknowledgement of that here. Could be brief and use the Docusaurus "note" syntax, like:

```
:::tip note
This example can currently only be expressed using this alternate syntax.
:::
```

Reply:
I'd initially thought this example could only be expressed using the alternate syntax, but that really isn't actually true since the inputs are coming from separate files.